OrbitGl/LiveFunctionsDataView.cpp

```diff
@@ -77,10 +77,12 @@ std::string LiveFunctionsDataView::GetValue(int row, int column) {
   }
 }
 
-void LiveFunctionsDataView::OnSelect(int row) {
+void LiveFunctionsDataView::OnSelect(std::optional<int> row) {
+  CHECK(row.has_value());
```

Comment:
This looks dangerous, I think the case `!row.has_value()` must be handled. It should be possible to not have any function selected, wasn't this handled before? 

Right now, wouldn't this happen if you hook a function, start a capture, and then unhook that function again and restart?

Reply:
Thanks for raising this issue. That's really a good question I should consider.

In the current version, we shouldn't reach the case that `!row.has_value()` for the LiveFunctionsDataView and the ProcessDataView. The reasons are the following:
- If it is an external selection (i.e., the selection made by the users when clicking rows in the data view panel), we always pass `currentIndex().rows()` to `OnSelect()` when `currentIndex().rows()` is valid. Therefore, for the external selections, we always pass a valid row to `OnRowSelected()` and `OnSelect()`.  
- But we do pass `std::nullopt` to `OnRowSelected()` and `OnSelect()` in **some internal selection cases**, which makes us able to clear the selection in the data view and process the no selection case. But we only do this for the `SamplingReportDataView` but not for the other two data views.

But you are right, this CHECK might cause crashes in the future. I will try to find a better way to handle this.